### PR TITLE
Fixing Issue with Definition ID

### PR DESCRIPTION
**Pull request by LibbaLawrence**
Definition ID bug fix, and Integer type fix for array.

### CODE REVIEW

File: packages/python-packages/protocol-stub-generator/setup.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
     author_email="azuresdkengsysadmins@microsoft.com",
     license="MIT License",
     packages=find_packages(),
-    install_requires=["astroid","pyyaml"],
+    install_requires=["astroid","pyyaml","re"],
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/setup.py:32]: `re` is part of the python standard library, no need to add it here.

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 2]
```diff
@@ -466,6 +469,7 @@ def from_yaml(cls, yaml_data: Dict[str, Any], op_group_num, op_num, namespace):
             ]["language"]["default"]["description"]
 
         return cls(
+            operation_group = yaml_data["operationGroups"][op_group_num]["language"]["default"]["name"],
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:472]: nit: no spaces between the assignment

```suggestion
            operation_group=yaml_data["operationGroups"][op_group_num]["language"]["default"]["name"],
```

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 3]
```diff
@@ -521,13 +525,13 @@ def add_first_line(self):
...
-        token.set_navigation_id(self.namespace + self.operation + "overview")
+        token.set_definition_id(self.namespace + self.operation_group + self.operation + "overview")
+        token.set_navigation_id(self.namespace + self.operation_group + self.operation + "overview")
         self.overview_tokens.append(token)
         self.add_keyword(
-            self.namespace + self.operation,
+            self.namespace + self.operation_group + self.operation,
             self.operation,
-            self.namespace + self.operation,
+            self.namespace + self.operation_group + self.operation,
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:534]: why do we need to pass the same value as a parameter twice to add_keyword?

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 4]
```diff
@@ -765,16 +773,16 @@ def get_type(data, page=False):
                 return_type = data["elementType"]["language"]["default"]["name"] + "[]"
             else:
                 return_type = data["elementType"]["language"]["default"]["name"]
-        if return_type == "number":
+        if "number" in return_type:
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:776]: judging by the code above, `return_type` is a string, but here it looks like we're expecting an iterable. I guess I'm wondering why the change? Should the others be changed too?

Comment (LibbaLawrence) [reply packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:776]: this was to catch if number was in any part of the return to replace it

Comment (kristapratico) [reply packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:776]: discussed offline

**Review by kristapratico — APPROVED** [review]
